lms/djangoapps/discussion/api/external_discussions.py

```diff
@@ -0,0 +1,44 @@
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from .data import ExternalDiscussionData
+from ..models import ExternalDiscussionsIdMapping
+
+
+def get_external_discussion_context(context_key: CourseKey, usage_key: UsageKey) -> ExternalDiscussionData:
+    try:
+        discussion_mapping = ExternalDiscussionsIdMapping.objects \
+            .get(context_key=context_key, usage_key=usage_key)
+    except ExternalDiscussionsIdMapping.DoesNotExist:
+        # Could happen if it hasn't been published.
+        raise ExternalDiscussionData.DoesNotExist(
+            "No ExternalDiscussionData for context_key={context_key} " \
+            "and usage_key={usage_key}".format(
+                context_key=context_key,
+                usage_key=usage_key
+            )
+        )
```

Comment:
return None instead of throwing exception.

Reply:
Done in next commit